types/src/transaction/mod.rs

```diff
@@ -289,6 +299,7 @@ impl RawTransaction {
         sender_public_key: Ed25519PublicKey,
```

Comment:
Ed25519 private keys contain their corresponding private key, so (if desired), you can simplify the API by omitting both `sender_public_key` and `secondary_public_keys` + using `Ed25519PublicKey::from(private_key)` in the function body.

Reply:
Interesting that we need to output the public key when we sign. Maybe in the future we could consider an api the outputs a tuple `(pubKey, sig)`, as the pubKey [is already computed internally to `sign`](https://github.com/diem/diem/blob/master/crypto/crypto/src/ed25519.rs#L109). In fact this makes signature generation a bit slower. I might take advantage of this observation and in the future add an api that takes pubKey as an input as well when signing to speed up signing by an expected 30%+. Thanks for the hint!

Reply:
Also @sblackshear note that in the original Dalek lib, [you provide a pubKey as input when you sign](https://github.com/dalek-cryptography/ed25519-dalek/blob/master/src/secret.rs#L390), which means that their signing might seem to return successfully, but the signature will fail on verification due to an accidental private-public key mismatch. That was the reason we originally had the private key as an input only (it's safer), but I might reconsider this now, because it's slower. Wdyt?